### PR TITLE
Add OpenGSN support to Baal

### PR DESCRIPTION
**Pull request by akshaan**
This change adds support for OpenGSN-style meta-transactions (EIP-2771) to Baal contract functions

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -44,6 +44,9 @@ function etherscan() {
 
 const config: HardhatUserConfig = {
   networks: {
+    hardhat: {
+      allowUnlimitedContractSize: true
```
Comment (akshaan) [inline hardhat.config.ts]: This was added to enable local testing with contracts larger than 24Kb. Not sure if we want to keep this around for future local testing, or get rid of it

Comment (dekanbro) [reply hardhat.config.ts]: we should be able to remove this now

**Comment by dekanbro** [comment]
On your note about using the params struct, we should not need it now. I was able to remove the name and symbol because they are being set in the summoner now

**Review by brossetti1 — APPROVED** [review]
if the build succeeds and the tests pass, this looks good to me, nice @akshaan !

**Review by dekanbro — APPROVED** [review]
we should be able to keep the destructering

**Pull request merged**